client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/QueryOperation.java

```diff
@@ -86,7 +86,7 @@ protected QueryResponse executeOperation(Transport transport) {
             EnumMarshaller encoder = (EnumMarshaller) remoteQuery.getSerializationContext().getMarshaller(value.getClass());
             value = encoder.encode((Enum) value);
          } else if (value instanceof Boolean) {
-            value = (Boolean) value ? 1 : 0;
+            value = value.toString();
```

Comment:
That too. And since it impacts index format I wanted to do it early. Right now it does not matter much, but it will if we decide later to expose the jpql string. We should ensure the same query string works both in embedded and remote.
ATM we still have enum and date which are handled in a particular way for protobuf ...